### PR TITLE
Migrate metadata from setup.py to setup.cfg

### PR DESCRIPTION
**Pull request by cclauss**
This pull request was created using https://pypi.org/project/setuptools-py2cfg plus manual modifications.

https://setuptools.pypa.io/en/latest/userguide/quickstart.html#basic-use
> We recommend users expose as much as possible configuration in a more declarative way via the [pyproject.toml](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html) or [setup.cfg](https://setuptools.pypa.io/en/latest/userguide/declarative_config.html), and keep the setup.py minimal with only the dynamic parts (or even omit it completely if applicable).

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,10 +1,61 @@
+[metadata]
+license_file = LICENSE
+name = flake8-bugbear
+version = attr: bugbear.__version__
```
Comment (cooperlees) [inline setup.cfg:4]: TIL - This will help releases. Thanks.

**Review by cooperlees — APPROVED** [review]
LGTM - Thanks ...

Just one question; why not move to `pyproject.toml`?
  - I'm happy to take this step than pyproject.toml but just wondering why
  - I believe setuptools support enough of the PEP now for pyproject.toml for pure python projects like us - I could be wrong tho

**Comment by cclauss** [comment]
> why not move to pyproject.toml? 

Because flake8 does not support py project.toml.  PyCQA/flake8#234

**Comment by cooperlees** [comment]
Good enough reason for now. Thanks for this. Lets roll from here and move to `pyproject.toml` when flake8 does.

**Comment by wwuck** [comment]
@cooperlees @cclauss FYI, there is a plugin that allows `pyproject.toml` configuration for flake8.

https://pypi.org/project/Flake8-pyproject/

**Comment by cclauss** [comment]
My sense is that a separate, supported text file is safer in the near-term and long-term than adding another dependency.

In fact, there are numerous outside dependencies that attempt to paper over this situation:
* https://pypi.org/project/Flake8-pyproject
* https://pypi.org/project/pflake8
* https://pypi.org/project/flake9
* https://pypi.org/project/flakehell
* https://pypi.org/project/ruff <-- Shows great promise

But none of these are from PyCQA and I believe that the best approach would be for the PyCQA to _solve_ this problem internally rather than encouraging the community to paper over it.

**Comment by cooperlees** [comment]
Totally agree with PyCQA solving this and making it first class.

Lets follow + comment on https://github.com/PyCQA/flake8/issues/234

**Pull request merged**